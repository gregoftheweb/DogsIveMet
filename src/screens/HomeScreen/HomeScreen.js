import React from 'react';
import {StyleSheet, Text, Image, View, useWindowDimensions} from 'react-native';
import CustomButton from '../../components/CustomButton';
import Logo from '../../../assets/images/cameraIcon.png';

const HomeScreen = () => {
  const {height} = useWindowDimensions();

  const onNewDogPressed = data => {
    console.log('Create new Dog Record');
    //navigation.navigate('SignIn');
  };

  return (
    <View>
      <Image
        source={require('../../../assets/images/DIM-Logo-v1.png')}
        style={[styles.logo, {height: height * 0.3}]}
        resizeMode="contain"
      />

      <Text>HomeScreen hello world</Text>
      <Text>Cool story bro!</Text>
      <CustomButton
        text="New Dog"
        onPress={onNewDogPressed}
        fgColor="white"
        type="PRIMARY"
        testID="buttonNewDog"
      />
    </View>
  );
};

export default HomeScreen;

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: '#fbf9fc',
    alignItems: 'center',
    width: '100%',
  },
  logo: {
    maxWidth: 200,
  },
});
