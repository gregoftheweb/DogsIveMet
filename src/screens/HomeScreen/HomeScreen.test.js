import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import HomeScreen from './HomeScreen';

//check homescreen
it('renders correctly', () => {
  renderer.create(<HomeScreen />);
});

//check onpress events
describe('HomeScreen', () => {
  test('check button press', () => {
    const component = renderer.create(<HomeScreen />);
    console.log = jest.fn();
    // Find the button that has props.testID === 'button'
    const button = component.root.findByProps({testID: 'buttonNewDog'});
    // All codes that causes state updates should wrap in act(...)

    button.props.onPress();

    // Expect the result
    expect(console.log).toHaveBeenCalledWith('Create new Dog Record');
  });
});
