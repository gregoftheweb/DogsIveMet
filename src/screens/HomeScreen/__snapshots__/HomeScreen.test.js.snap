// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen render Homescreen 1`] = `
<View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/images/DIM-Logo-v1.png",
      }
    }
    style={
      Array [
        Object {
          "maxWidth": 200,
        },
        Object {
          "height": 400.2,
        },
      ]
    }
  />
  <Text>
    HomeScreen hello world
  </Text>
  <Text>
    Cool story bro!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "marginVertical": 5,
          "padding": 15,
          "width": "80%",
        },
        Object {
          "width": 600,
        },
        Object {
          "backgroundColor": "#00749c",
          "fontWeight": "500",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontWeight": "600",
          },
          Object {
            "color": "#52082b",
            "fontFamily": "sans-serif",
            "fontSize": 25,
            "fontWeight": "bold",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      New Dog
    </Text>
  </View>
</View>
`;
